[MINOR UPDATE]: /docs update for DRILL-8117 (#2763)
apache/drill: docs/dev/ClusterFixture.md

Again, you can pass a Java value which the test code will convert to a string, then will build the `ALTER SESSION` command.

# Try-with-resource Style of Creating Single-use Client Fixtures.

> The benefit of Cluster Fixture framework is to define specific config for specific clusterFixture and clientFixture as needed flexibly.

Reviewer: ```suggestion
A benefit of the Cluster Fixture framework is the ability to define specific configs for specific clusterFixtures and clientFixtures as needed flexibly.
```

> In some cases, clusterFixture has been initialized, and we need to create several different config clients for different test cases,

Reviewer: ```suggestion
In some cases, clusterFixture has been initialized and we need to create several different config clients for different test cases.
```
```suggestion
In some cases, a clusterFixture has been initialized and we need to create several different config clients for different test cases.
```

> We could use try-with-resource style to creating single-use clientFixture.

Reviewer: ```suggestion
Using Java's try-with-resources syntax to create a single-use clientFixture is a convenient way to ensure that the clientFixture will automatically be closed once we've finished with it.
```

* The (local) file system location
* The default format

# Exception Matcher

> The `QueryBuilder` provides a clean and concise way to handle Exception match which includes type match and pattern match:

Reviewer: ```suggestion
The `QueryBuilder` provides a clean and concise way to handle UserException matching which includes error type matching and error message pattern matching:
```

```
    @Test
    public void unsupportedLiteralValidation() throws Exception {
      String query = "ALTER session SET `%s` = %s";

      client.queryBuilder()
        .sql(query, ENABLE_VERBOSE_ERRORS_KEY, "DATE '1995-01-01'")
        .userExceptionMatcher()
        .expectedType(ErrorType.VALIDATION)
        .include("Drill doesn't support assigning literals of type")
        .match();
    }
```
* Use `.userExceptionMatcher` to call UserExceptionMatcher
* Use `.expectedType` to define expected Error type

> * Use `.include` to define expected Error pattern

Reviewer: ```suggestion
* Use `.include` to define an expected error message regex pattern
* Use `.exclude` to define an unexpected error message regex pattern
```
